ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,244 @@
+                {
+                    if (array[j] > array[j+1])
+                    {
+                        int temp = array[j+1];
+                        array[j+1] = array[j];
+                        array[j] = temp; 
+                    }
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// BubbleSort algorithm in reverse order
+        /// </summary>
+        /// <param name="array">sort this array in reverse order</param>
+
+         static void BubbleSortDesc(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+            {
+                for (int j = 0; j < array.Length - i - 1; j++)
+                {
+                    if (array[j] < array[j + 1])
+                    {
+                        int temp = array[j + 1];
+                        array[j + 1] = array[j];
+                        array[j] = temp;
+                    }
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// InsertionSort algorithm in direct order
+        /// </summary>
+        /// <param name="array">sort this array in direct order</param>
+         static void InsertionSortAsc(int[] array)
+        {
+           
+            for (int i = 1; i < array.Length; i++)
+            {
+                int key = i;
+                while (key > 0 && array[key] < array[key - 1] )
+                {
+                    int temp = array[key-1];
+                    array[key - 1] = array[key];
+                    array[key] = temp;
+                    key--;
+                }
+            }
+        }
+
+        /// <summary>
+        /// InsertionSort algorithm in reverse order
+        /// </summary>
+        /// <param name="array">sort this array in reverse order</param>
+         static void InsertionSortDesc(int[] array)
+        {
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                int key = i;
+                while (key > 0 && array[key] > array[key - 1])
+                {
+                    int temp = array[key - 1];
+                    array[key - 1] = array[key];
+                    array[key] = temp;
+                    key--;
+                }
+            }
+        }
+
+        enum SortAlgorithmType
+        {
+            SelectionSort,
+            BubbleSort,
+            InsertionSort
+        }
+        enum OrderBy
+        {
+            Asc,
+            Desc
+        }
+
+
+        /// <summary>
+        /// Sort array in accordance with choosen algorithm and direction
+        /// </summary>
+        /// <param name="array">array to sort</param>
+        /// <param name="sortAlgorithmType">type of algorithm sort</param>
+        /// <param name="orderBy">Asc - direct order, Desc - reverse order</param>
+        static void SortArray( int[] array, SortAlgorithmType sortAlgorithmType, OrderBy orderBy)
+        {
+
+            if (orderBy == OrderBy.Asc)
```

Comment:
Again the next 2 blocks are almost identical, try to move sort order into methods and simplify your code

Reply:
I did it this way, not sure that it was exactly what you meant, but it helped to reduce the size of this method twice.